src/collection.js

```diff
@@ -537,6 +590,7 @@ export default class Collection {
         if (!syncResultObject.ok) {
           return syncResultObject;
         }
+        // XXX import of data should be done after conflict handling.
```

Comment:
Currently, we import the data *first* and then we check if we have conflicts. I would have expected to retrieve everything, check for conflicts (and solve them) and finally import.